great_expectations/execution_engine/split_and_sample/sparkdf_data_splitter.py

```diff
@@ -0,0 +1,288 @@
+        given month (if only month is chosen for date_parts) for ALL years.
+        This may be useful for viewing seasonality, but you can also specify
+        multiple date_parts to achieve date_trunc like behavior e.g.
+        year, month and day.
+
+        Args:
+            df: dataframe from batch data.
+            column_name: column in data used to determine split.
+            batch_identifiers: should contain a dateutil parseable datetime whose date parts
+                will be used for splitting or key values of {date_part: date_part_value}
+            date_parts: part of the date to be used for splitting e.g.
+                DatePart.DAY or the case-insensitive string representation "day"
+
+        Returns:
+            Dataframe with splitting applied.
+        """
+        self._validate_date_parts(date_parts)
+
+        date_parts: List[DatePart] = self._convert_date_parts(date_parts)
+
+        column_batch_identifiers: dict = batch_identifiers[column_name]
+
+        date_parts_dict: dict = (
+            self._convert_datetime_batch_identifiers_to_date_parts_dict(
+                column_batch_identifiers, date_parts
+            )
+        )
+
+        for date_part, date_part_value in date_parts_dict.items():
+            df = df.filter(
+                getattr(F, self._convert_date_part_to_spark_equivalent(date_part))(
+                    F.col(column_name)
+                )
+                == date_part_value
+            )
+        return df
+
+    @staticmethod
+    def _convert_date_part_to_spark_equivalent(date_part: [DatePart, str]) -> str:
+        """Convert the DatePart to a string representing the corresponding pyspark.sql.functions version.
+
+        For example DatePart.DAY -> pyspark.sql.functions.dayofmonth() -> "dayofmonth"
+
+        Args:
+            date_part: DatePart representing the part of the datetime to extract or string equivalent.
+
+        Returns:
+            String representing the spark function to use for the given DatePart.
+        """
+        date_part: DatePart = DatePart(date_part)
+
+        spark_date_part_decoder: dict = {
+            DatePart.YEAR: "year",
+            DatePart.MONTH: "month",
+            DatePart.WEEK: "weekofyear",
+            DatePart.DAY: "dayofmonth",
+            DatePart.HOUR: "hour",
+            DatePart.MINUTE: "minute",
+            DatePart.SECOND: "second",
+        }
+        return spark_date_part_decoder[date_part]
+
+    @staticmethod
+    def split_on_whole_table(
+        df,
+    ):
+        return df
+
+    @staticmethod
+    def split_on_column_value(df, column_name: str, batch_identifiers: dict):
+        return df.filter(F.col(column_name) == batch_identifiers[column_name])
+
+    @staticmethod
+    def split_on_converted_datetime(
+        df,
+        column_name: str,
+        batch_identifiers: dict,
+        date_format_string: str = "yyyy-MM-dd",
```

Comment:
Are there a finite number of options here?

Reply:
No - the possibilities are endless :) Well at least what is supported by spark.sql.functions.from_unixtime(). If you are thinking could we make this an enum, that would be great but I don't think so.
FYI - This is currently part of our API, I just moved it from the execution engine.